src/main/java/com/google/cloud/teleport/templates/PubsubToText.java

```diff
@@ -119,6 +119,31 @@
     @Default.String("5m")
     String getWindowDuration();
     void setWindowDuration(String value);
+  
+    @Description("The custom year pattern to use for the output directory.")
+    ValueProvider<String> getYearPattern();
+  
+    void setYearPattern(ValueProvider<String> yearPattern);
+  
+    @Description("The custom month pattern to use for the output directory.")
+    ValueProvider<String> getMonthPattern();
+  
+    void setMonthPattern(ValueProvider<String> monthPattern);
+  
+    @Description("The custom day pattern to use for the output directory.")
+    ValueProvider<String> getDayPattern();
+  
+    void setDayPattern(ValueProvider<String> dayPattern);
+  
+    @Description("The custom hour pattern to use for the output directory.")
+    ValueProvider<String> getHourPattern();
+  
+    void setHourPattern(ValueProvider<String> hourPattern);
+  
+    @Description("The custom minute pattern to use for the output directory.")
+    ValueProvider<String> getMinutePattern();
+  
+    void setMinutePattern(ValueProvider<String> minutePattern);
```

Comment:
I suggest moving all these options (+ the old output* options like `outputShardTemplate`) to a separate class (e.g. `WindowedFilenamePolicyOptions`), similar to how `StreamingOptions` are extended here, to avoid duplication in `PubsubToAvro` and `PubsubToText` templates. I'd put it in ` com.google.cloud.teleport.options` package.

Reply:
Makes sense. Will move these to a new `WindowedFilenamePolicyOptions` interface